File: hello-omp.f90
      program hello
      implicit none
      integer omp_get_thread_num, omp_get_num_threads
      write(6,*) 'outside of a parallel region, omp_num_threads=', 
     &                                  omp_get_num_threads()
C$OMP PARALLEL
      write(6,10) omp_get_thread_num(), omp_get_num_threads()
C$OMP END PARALLEL
  10  format('hello from thread ',i3, ' of ',i10)
      end

File: hello-omp.c
#include <stdio.h>
#if defined(NO_GETCPU)
int sched_getcpu( void ) { return -1; }
#else
#include <sched.h>
#endif


int main( int argc, char**argv )
{
    char host[1024];
    gethostname( host, 1024 );
#pragma omp parallel
    {
        printf( "Hello world from thread %d of %d running on cpu %2d on %s!\n", 
            omp_get_thread_num()+1, 
            omp_get_num_threads(),
            sched_getcpu(),
            host );
    }
    return 0;
}
